drpc, sql: enable TestRowLevelTTLJobMultipleNodes in DRPC mode

**Describe the problem**

`TestRowLevelTTLJobMultipleNodes` fails with DRPC on. First observed [here](https://github.com/cockroachdb/cockroach/issues/158317).

**To Reproduce**

Inconsistently repoducible (~every 2 in 3 fails):
`dev test pkg/sql/ttl/ttljob --filter TestRowLevelTTLJobMultipleNodes -v -- --test_env=COCKROACH_TEST_DRPC=true`

**Expected behavior**
Test should pass with DRPC enabled.

**Additional data / screenshots**
```
=== RUN   TestRowLevelTTLJobMultipleNodes/1_split
    test_server_shim.go:91: cluster virtualization disabled due to issue: #109391 (expected label: C-bug)
    test_server_shim.go:610: DRPC is enabled (override by TestingGlobalDRPCOption)
    ttljob_test.go:219: 
        	Error Trace:	pkg/sql/ttl/ttljob/ttljob_test.go:219
        	            				pkg/sql/ttl/ttljob/ttljob_test.go:740
        	Error:      	Should be true
        	Test:       	TestRowLevelTTLJobMultipleNodes/1_split
        	Messages:   	
        	            	expectedStatus="succeeded"
        	            	actualStatuses="failed"
        	            	 expectedError=""
        	            	  actualErrors="incorrect number of span partitions expected=2 actual=1
        	            	HINT: You have encountered an unexpected error.
        	            	
        	            	Please check the public issue tracker to check whether this problem is
        	            	already tracked. If you cannot find it there, please report the error
        	            	with details by creating a new issue.
        	            	
        	            	If you would rather not post publicly, please contact us directly
        	            	using the support form.
        	            	
        	            	We appreciate your feedback.
        	            	"
--- FAIL: TestRowLevelTTLJobMultipleNodes/1_split (1.79s)
=== RUN   TestRowLevelTTLJobMultipleNodes/2_splits
    test_server_shim.go:91: cluster virtualization disabled due to issue: #109391 (expected label: C-bug)
    test_server_shim.go:610: DRPC is enabled (override by TestingGlobalDRPCOption)
    ttljob_test.go:219: 
        	Error Trace:	pkg/sql/ttl/ttljob/ttljob_test.go:219
        	            				pkg/sql/ttl/ttljob/ttljob_test.go:740
        	Error:      	Should be true
        	Test:       	TestRowLevelTTLJobMultipleNodes/2_splits
        	Messages:   	
        	            	expectedStatus="succeeded"
        	            	actualStatuses="failed"
        	            	 expectedError=""
        	            	  actualErrors="incorrect number of span partitions expected=3 actual=1
        	            	HINT: You have encountered an unexpected error.
        	            	
        	            	Please check the public issue tracker to check whether this problem is
        	            	already tracked. If you cannot find it there, please report the error
        	            	with details by creating a new issue.
        	            	
        	            	If you would rather not post publicly, please contact us directly
        	            	using the support form.
        	            	
        	            	We appreciate your feedback.
        	            	"
--- FAIL: TestRowLevelTTLJobMultipleNodes/2_splits (1.87s)
```
Only fails for the `1_split` and `2_split` testcases, not for the `0_split` testcase.

**Environment:**
 - CockroachDB version [master]

Epic: CRDB-55382

Jira issue: CRDB-57269

--- a/pkg/sql/ttl/ttljob/ttljob_test.go
+++ b/pkg/sql/ttl/ttljob/ttljob_test.go
@@ -130,8 +130,6 @@ func newRowLevelTTLTestJobTestHelper(
 			Settings:          makeSettings(),
 			Knobs:             baseTestingKnobs,
 			InsecureWebAccess: true,
-			// TODO(server): re-enable DRPC once flakiness is addressed, see #158387.
-			DefaultDRPCOption: base.TestDRPCDisabled,
 		},
 	})
 	th.testCluster = testCluster

